tls/s2n_connection.c

```diff
@@ -675,6 +675,28 @@ int s2n_connection_get_cipher_preferences(struct s2n_connection *conn, const str
     return 0;
 }
 
+int s2n_connection_get_certificate_match(struct s2n_connection *conn, s2n_cert_sni_match *match_status)
+{
+    POSIX_ENSURE(conn, S2N_ERR_INVALID_ARGUMENT);
+    POSIX_ENSURE(match_status, S2N_ERR_INVALID_ARGUMENT);
+    POSIX_ENSURE(conn->mode == S2N_SERVER, S2N_ERR_CLIENT_MODE);
+
+    /* Server must have gotten past certificate selection */
+    POSIX_ENSURE(conn->handshake_params.our_chain_and_key, S2N_ERR_HANDSHAKE_NOT_COMPLETE);
```

Comment:
Won't this also occur in a PSK handshake? In this case I wonder if the "handshake not complete" error might be confusing. Maybe `S2N_ERR_NO_CERT_FOUND` could work?

Reply:
Oh ty thats a much better error